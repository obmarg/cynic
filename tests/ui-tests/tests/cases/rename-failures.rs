fn main() {}

mod schema {
    cynic::use_schema!("../../../schemas/starwars.schema.graphql");
}

#[derive(cynic::QueryFragment, Debug)]
#[cynic(schema_path = "../../../schemas/starwars.schema.graphql")]
struct Film {
    #[cynic(rename = "episode")]
    episode_id: Option<i32>,
}
