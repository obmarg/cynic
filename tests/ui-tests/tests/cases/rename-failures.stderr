error: Field episode does not exist on the GraphQL type Film. Did you mean created?
  --> $DIR/rename-failures.rs:10:13
   |
10 |     #[cynic(rename = "episode")]
   |             ^^^^^^

error: You can only alias a renamed field.  Try removing `alias` or adding a rename
  --> $DIR/rename-failures.rs:17:13
   |
17 |     #[cynic(alias)]
   |             ^^^^^
